### DISCUSSION TITLE
python程式語言和其他熱門語言的分析

### DISCUSSION THREAD

**Discussion by hapiwangy** [1101-GS4719-C/Python-Community-GS4719-C#398] [General]
接觸到新的語言，我很好奇他和其他語言為何差異那麼的大?用縮排來排版，不需要宣告資料型態，經過簡單的操作就可以使用很多功能強大的package。擁有如此多優點的程式語言難道就沒有缺點嗎?我找到的這兩篇文章讓我理解python在執行效能上和其他語言有差異的地方，也就是在那些強大功能背後所犧牲掉的東西。
[https://buzzorange.com/techorange/2018/08/14/python-is-slow/?fbclid=IwAR3-vIvrEVWNRcv8tST1jdtfoZW8HpH7GjFXC4iLYYkyT0sp4YAE0h7XXNY](https://buzzorange.com/techorange/2018/08/14/python-is-slow/?fbclid=IwAR3-vIvrEVWNRcv8tST1jdtfoZW8HpH7GjFXC4iLYYkyT0sp4YAE0h7XXNY)
[https://www.gushiciku.cn/pl/pKEr/zh-tw?fbclid=IwAR2xbfoHJwBvXzCbxa2nORbORKcL9pnYwsWFoOVNPaCTcmdE7TEgzYcvgiI](https://www.gushiciku.cn/pl/pKEr/zh-tw?fbclid=IwAR2xbfoHJwBvXzCbxa2nORbORKcL9pnYwsWFoOVNPaCTcmdE7TEgzYcvgiI)

**Comment by pekoranoinu** [comment]
看來我們不能只學python，學學其它程式語言會比較好。

**Reply by JunTingLin** [reply]
同意同意 可以看看Dcard上的這篇 [#請益 #問 如果只會python的話...](https://www.dcard.tw/f/softwareengineer/p/236339038)

**Reply by JunTingLin** [reply]
像我記得老師上課有說python有網頁框架可以開方網頁
但可能還要了解各種不同的知識 比如網頁的部屬 網路方面的知識
或甚至要跟後台Data base結合，就需要了解SQL語言
絕非單一打天下

**Comment by ThomasMeng0515** [comment]
Python確實在某些資料上處理速度較慢，像是Python中的列表(list)，而這也就是為什麼在應用機器學習時，會用到numpy將資料利用陣列的方式處理，這樣處理速度變會快10倍以上，而numpy的陣列處理速度會很快的原因便是因為可以整合C、C++的程式碼，才會使得處理速度遠高於Python中的列表。

**Comment by sean1325418** [comment]
以前是學C++，目前感覺語法跟python差很多，滿怕以後兩個會搞混

**Comment by 11timyyy** [comment]
現在同時學python和java，我也希望不要搞混😅

**Comment by ghost** [comment]
程式學習是真的容易混淆
而且不多加練習也都會生疏
建議學習有興趣的程式
日後依據職場或領域選擇其他程式語言學習
